proving-system/stark/src/lib.rs

```diff
@@ -21,316 +21,298 @@ pub type FE = FieldElement<PrimeField>;
 
 // DEFINITION OF CONSTANTS
 
-pub const ORDER_OF_ROOTS_OF_UNITY_TRACE: usize = 32;
-pub const ORDER_OF_ROOTS_OF_UNITY_FOR_LDE: usize = 1024;
-
 // We are using 3 as the offset as it's our field's generator.
-pub const COSET_OFFSET: u64 = 3;
+const COSET_OFFSET: u64 = 3;
+
+// TODO: change this to use more bits
+pub fn transcript_to_field<F: IsField>(transcript: &mut Transcript) -> FieldElement<F> {
+    let ret_value = transcript.challenge();
+    let ret_value_8: [u8; 8] = [
+        ret_value[0],
+        ret_value[1],
+        ret_value[2],
+        ret_value[3],
+        ret_value[4],
+        ret_value[5],
+        ret_value[6],
+        ret_value[7],
+    ];
+    let ret_value_u64 = u64::from_be_bytes(ret_value_8);
+    FieldElement::from(ret_value_u64)
```

Comment:
You could use this if it looks cleaner to you
```suggestion
    let value: u64 = u64::from_be_bytes(transcript.challenge()[..8].try_into().unwrap());
    FieldElement::from(value)
```